### PR TITLE
docs(document-search): create docs

### PR DESCRIPTION
**Pull request by konrad-czarnota-ds**

### CODE REVIEW

File: docs/how-to/document-search/use_rephraser.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```python
+from ragbits.core.llms.litellm import LiteLLM
+from ragbits.document_search.retrieval.rephrasers.llm import LLMQueryRephraser
+from ragbits.document_search.retrieval.rephrasers.prompts import QueryRephraserPrompt
+
+document_search = DocumentSearch(
+    query_rephraser=LLMQueryRephraser(LiteLLM("gpt-3.5-turbo"), QueryRephraserPrompt),
+    ...
+)
+results = await document_search.search("<query>")
```
Comment (kdziedzic68) [inline docs/how-to/document-search/use_rephraser.md:20]: maybe we should show how to use rephraser independently too ? if it is not intended to be used directly perhaps we should not include such components in public api

File: docs/how-to/document-search/use_reranker.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+```python
+import os
+from ragbits.document_search.retrieval.rerankers.litellm import LiteLLMReranker
+
+os.environ["COHERE_API_KEY"] = "<api_key>"
+
+document_search = DocumentSearch(
+    reranker=LiteLLMReranker("cohere/rerank-english-v3.0"),
+    ...
+)
```
Comment (kdziedzic68) [inline docs/how-to/document-search/use_reranker.md:21]: the same as with rephrasers

File: docs/how-to/document-search/use_reranker.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# How-To: Use Reranker
```
Comment (kdziedzic68) [inline docs/how-to/document-search/use_reranker.md:1]: higher level comment: maybe we should consider moving rerankers to ragbits core ? i could see some potential use cases of them in other tasks too like eg. data augmentation

Comment (konrad-czarnota-ds) [reply docs/how-to/document-search/use_reranker.md:1]: I think that this is a good topic for the grooming. This task was about creating docs to the components that are already there, they might be moved in the future but I wouldn't move functionalities between packages in doc creation task

Comment (kdziedzic68) [reply docs/how-to/document-search/use_reranker.md:1]: agree - just wanted to raise a topic here. comment is not directly related to PR itself. marking as resolved

**Review by kdziedzic68 — APPROVED** [review]

**Pull request merged**